internal/service/teams/resource_teams.go

```diff
@@ -61,36 +60,36 @@ func Resource() *schema.Resource {
 	}
 }
 
-func resourceMongoDBAtlasTeamCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	conn := meta.(*config.MongoDBClient).Atlas
+func resourceCreate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	connV2 := meta.(*config.MongoDBClient).AtlasV2
 	orgID := d.Get("org_id").(string)
 
 	// Creating the team
-	teamsResp, _, err := conn.Teams.Create(ctx, orgID,
-		&matlas.Team{
+	teamsResp, _, err := connV2.TeamsApi.CreateTeam(ctx, orgID,
+		&admin.Team{
 			Name:      d.Get("name").(string),
-			Usernames: conversion.ExpandStringListFromSetSchema(d.Get("usernames").(*schema.Set)),
-		})
+			Usernames: conversion.ExpandStringPointerListFromSetSchema(d.Get("usernames").(*schema.Set)),
```

Comment:
as above you can do before: 
usernames :=  conversion.ExpandStringPointerListFromSetSchema(d.Get("usernames").(*schema.Set))
and use &usernames here